### PR TITLE
feature: upgrade the androidx and get all tests passing

### PR DESCRIPTION
**Pull request by thomaszurkan-optimizely**
## Summary
- Upgraded to androidx.  Fixed all unit tests that either didn't compile or failed.
- Still need to update to the WorkManager and remove our job services.

## Test plan
- lots of manual testing as well as travis and unit tests.

## Issues

#350 #345

### CODE REVIEW

File: android-sdk/build.gradle [hunk 1]
```diff
@@ -69,18 +69,19 @@ dependencies {
     }
 
     compileOnly "com.fasterxml.jackson.core:jackson-databind:$jacksonversion"
-    implementation "com.android.support:support-annotations:$support_annotations_ver"
+    implementation "androidx.annotation:annotation:$annotations_ver"
+    implementation 'com.google.guava:guava:29.0-android'
```
Comment (yogurtearl) [inline android-sdk/build.gradle:73]: the jar for guava is 2.5MB, please don't add that as a dep. 😬 

it is not clear from the changes below why you would need this.

File: datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileServiceTest.java [hunk 2]
```diff
@@ -90,7 +92,8 @@ public void testBinding() throws TimeoutException {
 
 
         DatafileService datafileService = ((DatafileService.LocalBinder) binder).getService();
-        DatafileLoader datafileLoader = new DatafileLoader(datafileService, datafileClient, datafileCache, MoreExecutors.newDirectExecutorService(), mock(Logger.class));
+        DatafileLoader datafileLoader = new DatafileLoader(datafileService, datafileClient, datafileCache,
+                MoreExecutors.listeningDecorator(Executors.newSingleThreadExecutor()), mock(Logger.class));
```
Comment (yogurtearl) [inline datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileServiceTest.java:96]: I think just `Executors.newSingleThreadExecutor()` is fine here, no need for `MoreExecutors.listeningDecorator`

File: datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileServiceTest.java [hunk 3]
```diff
@@ -21,11 +21,12 @@
...
 import android.os.IBinder;
-import android.support.annotation.RequiresApi;
-import android.support.test.InstrumentationRegistry;
-import android.support.test.espresso.core.deps.guava.util.concurrent.ListeningExecutorService;
-import android.support.test.espresso.core.deps.guava.util.concurrent.MoreExecutors;
-import android.support.test.rule.ServiceTestRule;
+
+import androidx.annotation.RequiresApi;
+import androidx.test.espresso.core.internal.deps.guava.util.concurrent.ListeningExecutorService;
+import androidx.test.espresso.core.internal.deps.guava.util.concurrent.MoreExecutors;
```
Comment (yogurtearl) [inline datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileServiceTest.java:27]: using `.internal.` apis in espresso is dangerous, could break with any espresso upgrade

File: test-app/build.gradle [hunk 4]
```diff
@@ -49,8 +49,8 @@ dependencies {
 
     implementation "com.noveogroup.android:android-logger:$android_logger_ver"
     // implementation 'com.optimizely.ab:android-sdk:1.0.0'
-    implementation 'com.android.support:appcompat-v7:28.0.0'
-    implementation 'com.android.support:design:28.0.0'
+    implementation 'androidx.appcompat:appcompat:1.0.0'
+    implementation 'com.google.android.material:material:1.0.0'
```
Comment (yogurtearl) [inline test-app/build.gradle:53]: using `1.0.0` during the initial androidx upgrade is a good idea, but these are quite a bit out of date, so I recommend a quick follow on PR to upgrade all the androidx libs to the latest.

File: datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileLoaderTest.java [hunk 5]
```diff
@@ -110,7 +111,7 @@ public void loadFromCDNWhenNoCachedFile() throws MalformedURLException, JSONExce
 
     @Test
     public void loadWhenCacheFileExistsAndCDNNotModified() {
-        final ListeningExecutorService executor = MoreExecutors.newDirectExecutorService();
+        final ListeningExecutorService executor = MoreExecutors.listeningDecorator(Executors.newSingleThreadExecutor());
```
Comment (yogurtearl) [inline datafile-handler/src/androidTest/java/com/optimizely/ab/android/datafile_handler/DatafileLoaderTest.java:114]: unless you are calling `addListener` somewhere here, you can replace all the `listeningDecorator` stuff with just this:
```suggestion
        final ExecutorService executor = Executors.newSingleThreadExecutor();
```

**Review by msohailhussain — APPROVED** [review]
lgtm

**Pull request merged**